src/sql/platform/query/common/queryRunner.ts

```diff
@@ -135,39 +135,43 @@ export default class QueryRunner extends Disposable {
 	/**
 	 * Cancels the running query, if there is one
 	 */
-	public cancelQuery(): Thenable<azdata.QueryCancelResult> {
+	public cancelQuery(): Promise<azdata.QueryCancelResult> {
 		return this._queryManagementService.cancelQuery(this.uri);
 	}
 
 	/**
 	 * Runs the query with the provided query
 	 * @param input Query string to execute
 	 */
-	public runQuery(input: string, runOptions?: azdata.ExecutionPlanOptions): Thenable<void>;
+	public runQuery(input: string, runOptions?: azdata.ExecutionPlanOptions): Promise<void>;
 	/**
 	 * Runs the query by pulling the query from the document using the provided selection data
 	 * @param input selection data
 	 */
-	public runQuery(input: azdata.ISelectionData, runOptions?: azdata.ExecutionPlanOptions): Thenable<void>;
-	public runQuery(input, runOptions?: azdata.ExecutionPlanOptions): Thenable<void> {
-		return this.doRunQuery(input, false, runOptions);
+	public runQuery(input: azdata.ISelectionData, runOptions?: azdata.ExecutionPlanOptions): Promise<void>;
+	public runQuery(input: string | azdata.ISelectionData, runOptions?: azdata.ExecutionPlanOptions): Promise<void> {
+		if (types.isString(input)) {
+			return this.doRunQuery(input, false, runOptions);
+		} else {
+			return this.doRunQuery(input, false, runOptions);
```

Comment:
Interesting - why is this necessary? Shouldn't it be able to use the doRunQuery that takes in a string | azdata.ISelectionData directly?

Reply:
Its a weird behavior of the typescript compiler, even though the inputs are the same in terms of overloading it doesn't resolve it as being correct, its basically an unnecessary check to make the typescript compiler happy. It's definitely a bug in the compiler (although there are a lot of weird behavior like this that actually has some reasoning).